no more trailArgs for clearer error messages

jo-pol opened the pull request:
The vague 'can not parse trailArg' becomes:

```
[easy-stage-dataset] Error: Bad arguments for option 'EASY-bag': 'xxx' - file 'xxx' doesn't exist
```

Still any string is accepted by Conf for staged-digital-object-set

jo-pol committed:
no more trailArgs for clearer error messages

Now any string is accepted by Conf for staged-digital-object-set, but we can get:
"[easy-stage-dataset] Error: Bad arguments for option 'EASY-bag': 'xxx' - file 'xxx' doesn't exist"

jo-pol committed:
padding refactored

jo-pol commented:
@janvanmansum @GeorgiKhomeriki @PaulBoon @ekoi @vesaakerman please have a look

janvanmansum commented on src/main/scala/nl/knaw/dans/easy/stage/Conf.scala:
```diff
@@ -38,12 +39,12 @@ class Conf(args: Seq[String]) extends ScallopConf(args) {
     default = Some(false))
-  val bag = trailArg[File](
-    name = "EASY-bag",
+  val bag = opt[File](
```
@jo-pol Please make these trailArgs again. And also:
- `submission-timestamp`: default to "now"
- `urn` and `doi`: default to "not set"

janvanmansum commented on src/main/scala/nl/knaw/dans/easy/stage/Conf.scala:
```diff
@@ -7,20 +7,21 @@ import org.rogach.scallop.{singleArgConverter, ValueConverter, ScallopConf}
   printedName = "easy-stage-dataset"
-  val indent_____ = printedName.replaceAll(".", " ")
   version(s"$printedName v${Version()}")
+  private val padding = Array.fill(printedName.length)(' ').mkString("")
```
@jo-pol , @GeorgiKhomeriki suggested

```
printedName.map(_ => " ")
```

janvanmansum commented on src/main/scala/nl/knaw/dans/easy/stage/Conf.scala:
```diff
@@ -7,20 +7,21 @@ import org.rogach.scallop.{singleArgConverter, ValueConverter, ScallopConf}
            |
            |Options:
-           |""".stripMargin)
+           |""".stripMargin.replaceAll("_+",padding))
```
@jo-pol 
If you don't like my first solution, how about just naming the "padding" value `_________`; then the `replaceAll` is no longer necessary and it still looks almost the same.

jo-pol replied:
Though it slightly increases maintenance in case the length of the value for printed name increases, this proposal at least is better readable

jo-pol replied:
but it doesn't compile

janvanmansum replied:
OK, so you'll have to use the braces:

```
Usage:

$printedName -t <submission-timestamp> -u <urn> -d <doi> [ -o ] \\
${_________}    <EASY-bag> <staged-digital-object-set>
```

But still pretty neat, yes?

jo-pol committed:
trailArgs again

- logging gives additional info if the bag folder does not exist
- padding in the synopsis is still not ideal

jo-pol committed:
tiny change

jo-pol committed:
Dictionary constructor needed jsObject + further refactorings

janvanmansum commented on src/main/scala/nl/knaw/dans/easy/stage/EasyStageFileItem.scala:
```diff
@@ -0,0 +1,31 @@
+
+  def run(implicit s: FilItemSettings): Try[Unit] = {
+    log.debug(s"settings = $s")
+    ???
```
Not quite finished :)

jo-pol replied:
Accidentally committed and pushed :-(

janvanmansum replied:
Well, no harm done, you'll submit this story soon anyway.
